src/Resources/views/CRUD/list_outer_rows_mosaic.html.twig

```diff
@@ -73,12 +73,12 @@ This template can be customized to match your needs. You should only extends the
                      objectId="{{ admin.id(object) }}">
                     {% if admin.hasAccess('edit', object) and admin.hasRoute('edit') %}
                         <a class="mosaic-inner-link"
-                           href="{{ admin.generateUrl('edit', {'id' : object|sonata_urlsafeid(admin) }) }}">
+                           href="{{ admin.generateUrl('edit', {(admin.idParameter) : object|sonata_urlsafeid(admin) }) }}">
                             {{ mosaic_content }}
                         </a>
                     {% elseif admin.hasAccess('show', object) and admin.hasRoute('show') %}
                         <a class="mosaic-inner-link"
-                           href="{{ admin.generateUrl('show', {'id' : object|sonata_urlsafeid(admin) }) }}">
+                           href="{{ admin.generateUrl('show', {(admin.idParameter) : object|sonata_urlsafeid(admin) }) }}">
```

Comment:
this can be changed to not use the filter right?

Reply:
yep https://github.com/sonata-project/SonataAdminBundle/pull/7061